catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -158,7 +158,11 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
         }
         map = new HashMap<>();
         for (AttributeDescriptor attribute : metacard.getMetacardType().getAttributeDescriptors()) {
-            map.put(attribute.getName(), metacard.getAttribute(attribute.getName()));
+            Attribute metacardAttribute = metacard.getAttribute(attribute.getName());
```

Comment:
I guess it depends on your definition of _exist_. It could mean the metacard attribute it `null`, or the attribute _value_ is `null`, or both, which isn't very clear from the comment imho. But it could just be me too :-)